content/ri/memory-optimizations.md

```diff
@@ -281,23 +278,22 @@ Suppose you are using large [sets](https://redis.io/docs/data-types/sets/) to so
 - Check if an observation already appeared in the stream.
 - Find the fraction or the number of observations in the stream that are smaller or larger than a given value.
 
-and you are ready to trade accuracy with speed and memory usage, consider using one of the following probabilistic data structures. You can use:
+If you are ready to trade accuracy with speed and memory usage, consider using one of the following probabilistic data structures. You can use:
 
 - [HyperLogLog](https://redis.io/docs/data-types/hyperloglogs/) to estimate the number of unique observations in a set.
 - [Bloom filter or cuckoo filter](https://redis.io/docs/stack/bloom/) to check if an observation has already appeared in the stream. False positive matches are possible, but false negatives are not.
 - [t-digest](https://redis.io/docs/stack/bloom/) to estimate the fraction or the number of observations in the stream that are smaller or larger than a given value.
 
 Bloom filter and cuckoo filter require [RedisBloom](https://redis.com/modules/redis-bloom/).
 
-#### Trade offs
+#### Trade-offs
 
-Following are the tradeoffs of using HyperLogLog:
+Following are the trade-offs of using HyperLogLog:
 
 1. The results achieved from HyperLogLog are not 100% accurate; they have an approximate standard error of 0.81%.
 1. Hyperloglog only tells you the unique count. It cannot tell you the elements in the set.
 
-For example, if you want to maintain how many unique IP addresses made an API call today, HyperLogLog tells you
-`46966 unique IPs for today`.
+For example, if you want to maintain how many unique IP addresses made an API call today, HyperLogLog tells you `46966 unique IPs for today`.
 
 But if your desired result is `Show me those 46966 IP Addresses`, it cannot show you. To achieve that, you need to maintain all IP addresses in a set.
```

Comment:
Remove 'it cannot show you'.

But if you want to retrieve all the unique IP Addresses, you would need to maintain a set containing all these addresses.